daemon/info.go

```diff
@@ -153,25 +153,47 @@ func (daemon *Daemon) SystemInfo() (*types.Info, error) {
 }
 
 // SystemVersion returns version information about the daemon.
-func (daemon *Daemon) SystemVersion() types.Version {
+func (daemon *Daemon) SystemVersion(APIVersion string) types.Version {
+	kernelVersion := "<unknown>"
+	if kv, err := kernel.GetKernelVersion(); err != nil {
+		logrus.Warnf("Could not get kernel version: %v", err)
+	} else {
+		kernelVersion = kv.String()
+	}
+
 	v := types.Version{
+		Components: []types.ComponentVersion{
+			{
+				Name:    "Engine",
+				Version: dockerversion.Version,
+				Details: map[string]string{
+					"GitCommit":     dockerversion.GitCommit,
+					"ApiVersion":    APIVersion,
```

Comment:
Still confused why we can't access `api.DefaultVersion` here directly

Reply:
I assumed a circular dependency, but it's true i didnt try.